androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/SurveyDetailScreen.kt

```diff
@@ -0,0 +1,47 @@
+package co.nimblehq.avishek.phong.kmmic.android.ui.screen.surveydetail
+
+import androidx.compose.foundation.pager.*
+import androidx.compose.material.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import co.nimblehq.avishek.phong.kmmic.android.ui.theme.ApplicationTheme
+import co.nimblehq.avishek.phong.kmmic.presentation.module.HomeViewModel
+import org.koin.androidx.compose.getViewModel
+
+@Composable
+fun SurveyDetailScreen(
+    homeViewModel: HomeViewModel = getViewModel(),
+    surveyId: String,
+    onBackClick: () -> Unit,
+) {
+    val viewState by homeViewModel.viewState.collectAsStateWithLifecycle()
+    val surveyUiModel = viewState.surveys.find { it.id == surveyId }
+
+    surveyUiModel?.let {
+        SurveyStartContent(surveyUiModel = it,
+            onStartClick = {
+                // TODO: implement in the integrate task
+            },
+            onBackClick = onBackClick
+        )
+    }
+}
+
+@Preview(showSystemUi = true)
+@Composable
+fun SurveyDetailScreenPreview(
```

Comment:
I am having trouble with preview for this screen. Can you check? I think we need to separate the screen content from the main screen and preview the content.

Reply:
I've updated the implementation in PR #120 to facilitate the previews. Could you please check that one so that we can avoid the rework?